include/alp/reference/blas1.hpp

```diff
@@ -2923,15 +2930,16 @@ namespace alp {
 		const Vector< InputType, InputStructure, Density::Dense, InputView, InputImfR, InputImfC, backend > &y,
 		const Ring &ring = Ring(),
 		const typename std::enable_if<
-			std::is_floating_point< OutputType >::value,
-		void >::type * const = nullptr
+			std::is_floating_point< OutputType >::value || grb::utils::is_complex< OutputType >::value,
+			void
+		>::type * const = nullptr
 	) {
```

Comment:
Similar comment as for `dot`.

Reply:
same. If you I can remove c++ scalar versions.